src/magnify/apps/core/serializers/rooms.py

```diff
@@ -106,4 +92,18 @@ def to_representation(self, instance):
             )
         output["is_administrable"] = is_admin
 
+        if instance.is_public or (
+            user.is_authenticated
+            and (
+                instance.user_accesses.filter(user=user)
+                or instance.group_accesses.filter(
+                    Q(group__members=user) | Q(group__administrators=user)
+                )
+            )
```

Comment:
```suggestion
            user.is_authenticated
            and instance.has_access)
```
it's only be moved, but it could be handy to add directly a method in the Room model so it can be reused, for instance
def has_access(user):

Reply:
Good idea :star_struck: :pray: